packages/core/i18n/core-i18n-server-internal/src/routes/translations.ts

```diff
@@ -6,54 +6,81 @@
  */
 
-import { createHash } from 'crypto';
 import { i18n } from '@kbn/i18n';
 import { schema } from '@kbn/config-schema';
 import type { IRouter } from '@kbn/core-http-server';
 
+const MINUTE = 60;
+const HOUR = 60 * MINUTE;
+const DAY = 24 * HOUR;
+
 interface TranslationCache {
   translations: string;
   hash: string;
 }
 
-export const registerTranslationsRoute = (router: IRouter, locale: string) => {
+export const registerTranslationsRoute = ({
+  router,
+  locale,
+  translationHash,
+  isDist,
+}: {
+  router: IRouter;
+  locale: string;
+  translationHash: string;
+  isDist: boolean;
+}) => {
   let translationCache: TranslationCache;
 
-  router.get(
-    {
-      path: '/translations/{locale}.json',
-      validate: {
-        params: schema.object({
-          locale: schema.string(),
-        }),
-      },
-      options: {
-        access: 'public',
-        authRequired: false,
-      },
-    },
-    (ctx, req, res) => {
-      if (req.params.locale.toLowerCase() !== locale.toLowerCase()) {
-        return res.notFound({
-          body: `Unknown locale: ${req.params.locale}`,
-        });
-      }
-      if (!translationCache) {
-        const translations = JSON.stringify(i18n.getTranslation());
-        const hash = createHash('sha1').update(translations).digest('hex');
-        translationCache = {
-          translations,
-          hash,
-        };
-      }
-      return res.ok({
-        headers: {
-          'content-type': 'application/json',
-          'cache-control': 'must-revalidate',
-          etag: translationCache.hash,
+  ['/translations/{locale}.json', `/translations/${translationHash}/{locale}.json`].forEach(
+    (routePath) => {
+      router.get(
+        {
+          path: routePath,
+          validate: {
+            params: schema.object({
+              locale: schema.string(),
+            }),
+          },
+          options: {
+            access: 'public',
+            authRequired: false,
+          },
         },
-        body: translationCache.translations,
-      });
+        (ctx, req, res) => {
+          if (req.params.locale.toLowerCase() !== locale.toLowerCase()) {
+            return res.notFound({
+              body: `Unknown locale: ${req.params.locale}`,
+            });
+          }
+          if (!translationCache) {
+            const translations = JSON.stringify(i18n.getTranslation());
+            translationCache = {
+              translations,
+              hash: translationHash,
+            };
+          }
+
+          let headers: Record<string, string>;
+          if (isDist) {
+            headers = {
+              'content-type': 'application/json',
+              'cache-control': `public, max-age=${365 * DAY}, immutable`,
+            };
+          } else {
+            headers = {
+              'content-type': 'application/json',
+              'cache-control': 'must-revalidate',
+              etag: translationCache.hash,
+            };
```

Comment:
I kept the `etag` cache control for non-distribution build, even if I'm not sure how useful this is, given the translation hash will change when translations are added/removed/changed.